polaris/hub/polarisfs.py

```diff
@@ -44,17 +45,28 @@ def __init__(
         polaris_client: "PolarisHubClient",
         dataset_owner: str,
         dataset_name: str,
+        use_listings_cache: bool = True,
+        listings_expiry_time=None,
+        max_paths=None,
         **kwargs: dict,
     ):
         super().__init__(**kwargs)
 
+        self.use_listings_cache = use_listings_cache
+        if self.use_listings_cache:
+            self.dircache = DirCache(listings_expiry_time=listings_expiry_time, max_paths=max_paths)
+        else:
+            self.dircache = None
+
```

Comment:
We can create the dircache through the parent class, see e.g. [here](https://github.com/fsspec/filesystem_spec/blob/518984d494e48f007d8aa48de882ed7014da2574/fsspec/spec.py#L149).

I also wouldn't let people configure it. Everyone should use this - It's faster!
```suggestion
    ):
        super().__init__(
            use_listing_cache=True, 
            listings_expiry_time: None,
            max_paths: None,
       )
```